### PR TITLE
feat(crd): add KongTarget CRD

### PR DESCRIPTION
**Pull request by randmonkey**
**What this PR does / why we need it**:
Add `KongTarget` CRD for Konnect entities. Also fixed some copy-pasta mistakes.

**Which issue this PR fixes**

Required for https://github.com/Kong/gateway-operator/issues/576.

**Special notes for your reviewer**:
Since targets are always attached to an upstream, here `upstreamRef` is required. Should we define it as required (for possibility that a `KongTarget` is attached to another upstream that is not generated from `KongUpstream`?)

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [ ] the `CHANGELOG.md` release notes have been updated to reflect significant changes

### CODE REVIEW

File: api/configuration/v1alpha1/kongupstream_types.go
Comment (pmalek) [inline api/configuration/v1alpha1/kongupstream_types.go:1]: 👍

File: api/configuration/v1alpha1/kong_target_types.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+	t.Status.Conditions = conditions
+}
+
+type KongTargetSpec struct {
+	// ControlPlaneRef is a reference to a ControlPlane this KongTarget is associated with.
+	// +optional
+	ControlPlaneRef *ControlPlaneRef `json:"controlPlaneRef,omitempty"`
+	// UpstreamRef is a reference to a KongUpstream this KongTarget is attached to.
+	// +required
+	UpstreamRef *TargetRef `json:"upstreamRef"`
```
Comment (pmalek) [inline api/configuration/v1alpha1/kong_target_types.go:110]: If this is required I believe we can change the type to be non-pointer, WDYT?

**Review by pmalek — COMMENTED** [review]
How about adding a bunch of UTs like in https://github.com/Kong/kubernetes-configuration/blob/960b674e12d69aa0ca2dd6be26c840bdf5a31768/test/unit/kongupstream_test.go and CRD validation tests in https://github.com/Kong/kubernetes-configuration/tree/main/test/crdsvalidation ?

File: api/configuration/v1alpha1/kong_target_types.go (hunk 1 above)
Comment (randmonkey) [reply api/configuration/v1alpha1/kong_target_types.go]: Done.

**Comment by randmonkey** [comment]
> How about adding a bunch of UTs like in https://github.com/Kong/kubernetes-configuration/blob/960b674e12d69aa0ca2dd6be26c840bdf5a31768/test/unit/kongupstream_test.go and CRD validation tests in https://github.com/Kong/kubernetes-configuration/tree/main/test/crdsvalidation ?

Added unit tests.

**Review by pmalek — APPROVED** [review]

**Pull request merged**